cli/src/commands/commit.rs

```diff
@@ -61,8 +64,22 @@ pub(crate) struct CommitArgs {
     )]
     tool: Option<String>,
     /// The change description to use (don't open editor)
-    #[arg(long = "message", short, value_name = "MESSAGE")]
+    #[arg(
+        long = "message",
+        short,
+        value_name = "MESSAGE",
+        conflicts_with = "stdin"
+    )]
     message_paragraphs: Vec<String>,
+    /// Read the change description from stdin
+    #[arg(long, conflicts_with = "interactive", conflicts_with = "tool")]
+    stdin: bool,
```

Comment:
Does `--stdin` conflicts with `--interactive`/`--tool` because the diff editor may be console-based? Since GUI tool wouldn't have such restriction, it might be better to unblock `--stdin --interactive`.